iOSMoviesApp/Controllers/DetailsPage/DetailsScreenViewController.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  DetailsScreenViewController.swift
+//  iOSMoviesApp
+//
+//  Created by Yash Agrawal on 11/07/25.
+//
+
+import Foundation
+import UIKit
+
+protocol DetailsScreenViewControllerDelegate: AnyObject {
+    func reloadTableData()
+}
+
+class DetailsScreenViewController: UIViewController {
+
+    @IBOutlet weak var tableView: UITableView!
+    
+    private var detailsVM : DetailsScreenViewModel!
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setupUI()
+        getDetails()
+    }
+    
+    private func setupUI() {
+        setupTableView()
+    }
+    
+    func setupMovie(movieVM : MovieViewModel){
+        self.detailsVM = DetailsScreenViewModel(delegate: self)
+        self.detailsVM.movie = movieVM
+    }
+    
+    private func getDetails() {
+        detailsVM.fetchCastDetails()
+        detailsVM.fetchMovieInfo()
+    }
+    
+    @IBAction func backToDiscovery(_ sender: UIBarButtonItem) {
+        self.navigationController?.popViewController(animated: true)
+    }
+}
+
+extension DetailsScreenViewController: UITableViewDataSource , UITableViewDelegate {
+    
+    private func setupTableView() {
+        let castNib: UINib = UINib(nibName: "CastDetailsTableViewCell", bundle: nil)
+        let reviewNib: UINib = UINib(nibName: "ReviewTableViewCell", bundle: nil)
+        let infoNib: UINib = UINib(nibName: "InfoTableViewCell", bundle: nil)
+        
```

Comment:
```suggestion
                self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: "DefaultCell")
```
You can register default cell here.
This will help you to remove redundant code.

Reply:
Done